src/core/probe/kernel/mod.rs

```diff
@@ -109,6 +134,18 @@ trait ProbeBuilder {
     fn attach(&mut self, target: &str) -> Result<()>;
 }
 
+fn reuse_map_fds(open_obj: &libbpf_rs::OpenObject, map_fds: &Vec<(String, i32)>) -> Result<()> {
+    if !map_fds.is_empty() {
```

Comment:
I noticed in more than one place some logic including is_empty() before iterating.
Is the purpose of this to avoid some hidden costs, e.g. into_iter() or something?